src/extension.ts

```diff
@@ -22,10 +22,38 @@ const autoExecute = () => {
   }
 };
 
+
+function changedEditor(e: any) {
+  let folders = workspace.workspaceFolders;
+  let terms = window.terminals;
+  if (folders != undefined) {
```

Comment:
Checking if the path includes the terminal name should work in most cases but can cause some issues in other cases. You can use `workspace.getWorkspaceFolder(e.document.uri)` to get the workspace folder and compare its name with the names of the terminals.